File: 2665-minimum-time-to-repair-cars/solution.cpp
#define ll long long
class Solution {
public:
    long long repairCars(vector<int>& ranks, int cars) {
        long long left = 0;
        long long right = 1e14;
        while (left < right) {
            ll mid = (left + right) / 2;
            ll count = 0;
            for (int i = 0; i < ranks.size(); i++) {
                count += sqrt(mid / ranks[i]);
            }
            if (count >= cars)
                right = mid;
            else
                left = mid + 1;
        }
        return right;
    }
};

File: 2665-minimum-time-to-repair-cars/solution.java
class Solution {
    public long repairCars(int[] ranks, int cars) {
        long l=0;
        long r=(long)1e14;
        while(l<r){
           long mid=(l+r)/2;
           long count=0;
           for(int rank:ranks){
            count+=(long)Math.sqrt((double)mid/rank);
           } 
           System.out.println("Count :"+count);
           System.out.println("Mid   :"+mid);
           System.out.println();
           if(count>=cars){
                r=mid;
           }else{
            l=mid+1;
           }
        }
        return l;
    }
}
